File: GestionIntApi/DTO/CreditoDTO.cs
namespace GestionIntApi.DTO
{
    public class CreditoDTO
    {

        public int Id { get; set; }

        public decimal Monto { get; set; }
        public int PlazoCuotas { get; set; }
        public string FrecuenciaPago { get; set; }
        public DateTime DiaPago { get; set; }

        public decimal ValorPorCuota { get; set; }
        public decimal TotalPagar { get; set; }
        public DateTime ProximaCuota { get; set; }
        public string ProximaCuotaStr { get; set; }
        public string Estado { get; set; }

       // public DateTime FechaCreacion { get; set; }
        public int ClienteId { get; set; }
    }
}

File: lib/models/credito_dto.dart
class CreditoDTO {
  int id;
  double monto;
  double entrada; // <--- Agregado
  int plazoCuotas;
  String frecuenciaPago;
  DateTime diaPago;
  double valorPorCuota;
  double totalPagar;
  DateTime proximaCuota;
  String? proximaCuotaStr;
  String? estado;
  int clienteId;

  CreditoDTO({
    this.id = 0,
    required this.monto,
    this.entrada = 0.0,
    required this.plazoCuotas,
    required this.frecuenciaPago,
    required this.diaPago,
    required this.valorPorCuota,
    required this.totalPagar,
    required this.proximaCuota,
    this.proximaCuotaStr,
    this.estado,
    this.clienteId = 0,
  });

  Map<String, dynamic> toJson() => {
    'Id': id,
    'Monto': monto,
    'Entrada': entrada, // Nuevo campo
    'PlazoCuotas': plazoCuotas,
    'FrecuenciaPago': frecuenciaPago,
    'DiaPago': diaPago.toIso8601String(),
    'ValorPorCuota': valorPorCuota,
    'TotalPagar': totalPagar,
    'ProximaCuota': proximaCuota.toIso8601String(),
    'ProximaCuotaStr': proximaCuotaStr,
    'Estado': estado,
    'ClienteId': clienteId,
  };
}